include/envoy/filesystem/filesystem.h

```diff
@@ -13,20 +14,31 @@
 namespace Envoy {
 namespace Filesystem {
 
+typedef std::bitset<4> FlagSet;
+
+static constexpr FlagSet DEFAULT_FLAGS_SET{0xf};
+
 /**
  * Abstraction for a basic file on disk.
  */
 class File {
 public:
   virtual ~File() = default;
 
+  enum Operation {
+    Read,
+    Write,
+    Create,
+    Append,
+  };
+
   /**
-   * Open the file with O_RDWR | O_APPEND | O_CREAT
+   * Open the file with Flag
    * The file will be closed when this object is destructed
    *
    * @return bool whether the open succeeded
    */
-  virtual Api::IoCallBoolResult open() PURE;
+  virtual Api::IoCallBoolResult open(const FlagSet flags = DEFAULT_FLAGS_SET) PURE;
```

Comment:
We don't generally allow default parameters on interface functions, so I would make it required or make a new open function that allows for specifying the open flags.